platform/Callback.h

```diff
@@ -30,6 +30,13 @@
 // MBED_CONF_PLATFORM_CALLBACK_NONTRIVIAL - support storing non-trivial function objects
 // MBED_CONF_PLATFORM_CALLBACK_COMPARABLE - support memcmp comparing stored objects (requires zero padding)
 
+#ifdef __ICCARM__
+/* Force callback.non-trivial for IAR */
+#undef MBED_CONF_PLATFORM_CALLBACK_NONTRIVIAL
+#define MBED_CONF_PLATFORM_CALLBACK_NONTRIVIAL 1
```

Comment:
Redefining config options is icky, so I'd rather not. Can be confusing for any third parties (people or code) checking the option, potentially without looking or including here.

But I don't have a better suggestion. Don't think we can have a "toolchain override" for the JSON. I can live with this.